src/models/document-content.ts

```diff
@@ -106,17 +106,18 @@ export const DocumentContentModel = types
       const action = o.action || LogEventName.CREATE_TILE;
       Logger.logTileEvent(action, tile, o.loggingMeta);
 
-      return { row, tile };
+      return { rowId: row.id, tileId: tile.id };
     },
   }))
   .actions((self) => ({
     addGeometryTile(addSidecarNotes?: boolean) {
       const result = self.addTileInNewRow(defaultGeometryContent(),
                                           { rowHeight: kGeometryDefaultHeight });
-      const { row } = result;
+      const { rowId } = result;
+      const row = self.rowMap.get(rowId);
       const tile = ToolTileModel.create({ content: defaultTextContent() });
       self.tileMap.put(tile);
-      row.insertTileInRow(tile, 1);
+      row!.insertTileInRow(tile, 1);
```

Comment:
Is it worth just returning `row` in `addTileInNewRow` to avoid the `!`?

Reply:
I had it returning { row, tile } originally, but ran into a situation (with selecting the newly created tile) in which MST was throwing warnings about accessing deleted objects. I decided it was safer to deal in IDs than ephemeral objects.